srcpkgs/chromium/template

```diff
@@ -242,29 +280,66 @@ do_configure() {
 		ppc64*) conf+=( 'target_cpu="ppc64"' ) ;;
 	esac
 
+	if [ "$CROSS_BUILD" ]; then
+		case "${XBPS_MACHINE}" in
+			x86_64*) conf+=( 'host_cpu="x64"' ) ;;
+			i686*) conf+=( 'host_cpu="x86"' ) ;;
+			arm*) conf+=( 'host_cpu="arm"' ) ;;
+			aarch64*) conf+=( 'host_cpu="arm64"' ) ;;
+			ppc64*) conf+=( 'host_cpu="ppc64"' ) ;;
+		esac
+	fi
+
 	if [ "$build_option_clang" ]; then
+		local gcc_version=$(gcc -dumpversion)
+		local clang_version=$(clang -dumpversion)
 		export CC=clang
 		export CXX=clang++
 		export AR=llvm-ar
 		export NM=llvm-nm
 		export CFLAGS="-Wno-unknown-warning-option"
 		export CXXFLAGS="-Wno-unknown-warning-option"
+		export BUILD_CC=clang
+		export BUILD_CXX=clang++
+		export BUILD_AR=llvm-ar
+		export BUILD_NM=llvm-nm
+		export BUILD_CFLAGS="-Wno-unknown-warning-option"
+		export BUILD_CXXFLAGS="-Wno-unknown-warning-option"
+		if [ "$CROSS_BUILD" ]; then
+			CFLAGS+=" --gcc-toolchain=/usr --sysroot=${XBPS_CROSS_BASE} -nostdinc -isystem ${XBPS_CROSS_BASE}/usr/include -isystem /usr/lib/clang/${clang_version}/include"
+			CXXFLAGS+=" --gcc-toolchain=/usr --sysroot=${XBPS_CROSS_BASE} -nostdinc++ -isystem ${XBPS_CROSS_BASE}/usr/include/c++/${gcc_version%.*} -isystem ${XBPS_CROSS_BASE}/usr/include/c++/${gcc_version%.*}/${XBPS_CROSS_TRIPLET} -isystem ${XBPS_CROSS_BASE}/usr/include/c++/${gcc_version%.*}/backward -nostdinc -isystem ${XBPS_CROSS_BASE}/usr/include -isystem /usr/lib/clang/${clang_version}/include"
+			export LDFLAGS="--gcc-toolchain=/usr --sysroot=${XBPS_CROSS_BASE}"
+		fi
 	fi
```

Comment:
Is deduplicating this block possible?

Reply:
We can't set it at the toplevel because the I think xbps-src adds its own CFLAGS etc, we theoretically don't need it in do_build, but it makes working on the build a whole lot easier since ninja might need to rebuild some ninja files.

Reply:
Maybe it could be moved into a function like `_clang_env` which is called from both.

Reply:
Sounds reasonable to me.